DirectProgramming/DPC++FPGA/Tutorials/Features/mem_channels/README.md

```diff
@@ -0,0 +1,310 @@
+of resources required to implement a DPC++ FPGA design.
+
+By default, the Intel® oneAPI DPC++ compiler configures each global memory type
+in a burst-interleaved manner where memory words are interleaved across the
+available memory channels. This usually leads to better throughput because it
+prevents load imbalance by ensuring that memory accesses do not favor one
+external memory channel over another. Moreover, burst-interleaving enables you
+to view a multi-channel memory as one large channel so you can write your code
+without worrying about where each buffer should be allocated. However, this
+configuration can be expensive in terms of FPGA resources because the global
+memory interconnect required to orchestrate the memory accesses across all the
+channels is complex. For more information about burst-interleaving, please
+refer to the [oneAPI DPC++ FPGA Optimization
+Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide).
+
+The Intel® oneAPI DPC++ compiler allows you to avoid this area overhead by
+disabling burst-interleaving and assigning buffers to individual channels. There
+are two advantages to such configuration:
+1. A simpler global memory interconnect is built, which requires a smaller
+   amount of FPGA resources than the interconnect needed for the
+   burst-interleaving configuration.
+2. Potential improvements to the global memory bandwidth utilization due to
+   less contention at each memory channel.
+
+Burst-interleaving should only be disabled in situations where satisfactory
+load balancing can be achived by assigning buffers to individual channels.
+Otherwise, the global memory bandwidth utilization may be reduced, which will
+negatively impact the throughput of your design. 
+
+To disable burst-interleaving, you need to pass the `-Xsno-interleaving` flag
+to your `dpcpp` command. With interleaving disabled, you can now specify, using
+the `mem_channel` property, in which memory channel each buffer should be
+allocated:
+```c++
+buffer a_buf(a_vec, {property::buffer::mem_channel{1}});
+buffer b_buf(b_vec, {property::buffer::mem_channel{2}});
+```
+The channels are assumed to have IDs `1,2,...,N` where `N` is the number of
+available channels.
+
+For FPGA boards that have multiple memory types, it is possible to select which
+memory you want to disable burst-interleaving for by passing the memory type to
+the `-Xsno-interleaving` flag: `-Xsno-interleaving=<global_memory_type>`. The
+memory type is usually indicated in the board specification XML file.
+
+An important observation is that the `mem_channel` property is a runtime
+specification which means you can change the selected channel ID for each
+buffer without having to recompile the device code. In other words, when you
+change the channel IDs, you only need to recompile the host code since
+`mem_channel` has no impact on the RTL generated by the compiler.
```

Comment:
Add a reference to the fast_recompile tutorial here, as a reference for how to recompile only host code?